pkg/util/workloadspread/workloadspread.go

```diff
@@ -61,3 +130,396 @@ func VerifyGroupKind(ref *appsv1alpha1.TargetReference, expectedKind string, exp
+
+	wsClone := matchedWS.DeepCopy()
+	var refresh, changed, isInject bool
+	var suitableSubset *appsv1alpha1.WorkloadSpreadSubsetStatus
+	var generatedUID string
+	var injectErr error
+
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		var err error
+		if refresh {
+			if err = h.Client.Get(context.TODO(), client.ObjectKey{Namespace: matchedWS.Namespace, Name: matchedWS.Name}, wsClone); err != nil {
+				klog.Errorf("error getting updated WorkloadSpread(%s/%s) from client", matchedWS.Namespace, matchedWS.Name)
+				return err
+			}
+		}
+
+		// check whether WorkloadSpread has suitable subset for the pod
+		// 1. changed indicates whether workloadSpread status changed
+		// 2. isInject indicates whether to inject workloadSpread subset in pod
+		// 3. suitableSubset is matched subset for the pod
+		changed, isInject, suitableSubset, generatedUID, injectErr = h.updateSubsetForPod(wsClone, pod, injectWS, operation)
+		if injectErr != nil || !changed {
+			return nil
+		}
+
+		// update WorkloadSpread status
+		if err = h.Client.Status().Update(context.TODO(), wsClone); err != nil {
+			refresh = true
+		} else {
+			klog.V(3).Infof("update workloadSpread(%s/%s) SubsetStatus(%s) missingReplicas(%d) creatingPods(%d) deletingPods(%d) success",
+				wsClone.Namespace, wsClone.Name, suitableSubset.Name,
+				suitableSubset.MissingReplicas, len(suitableSubset.CreatingPods), len(suitableSubset.DeletingPods))
+		}
+		return err
+	}); err != nil {
+		klog.Errorf("update WorkloadSpread(%s/%s) error %s", matchedWS.Namespace, matchedWS.Name, err.Error())
+		return err
+	}
+
+	// if create pod, inject affinity、toleration、metadata in pod object
+	if isInject {
+		if _, injectErr = injectWorkloadSpreadIntoPod(matchedWS, pod, suitableSubset.Name, generatedUID); injectErr != nil {
+			klog.Errorf("failed to inject Pod(%s/%s) subset(%s) data for workloadSpread(%s/%s)",
+				pod.Namespace, pod.Name, suitableSubset.Name, matchedWS.Namespace, matchedWS.Name)
+			return injectErr
+		}
+		klog.V(3).Infof("inject Pod(%s/%s) subset(%s) data for workloadSpread(%s/%s)",
+			pod.Namespace, pod.Name, suitableSubset.Name, matchedWS.Namespace, matchedWS.Name)
+	}
+
+	klog.V(3).Infof("handler Pod(%s/%s) generatedUID(%s) operation(%s) for workloadSpread(%s/%s) done",
+		pod.Namespace, podName, generatedUID, operation, matchedWS.Namespace, matchedWS.Name)
+
+	return injectErr
+}
+
+// return four parameters:
+// 1. changed(bool) indicates if workloadSpread.Status has changed
+// 2. inject(bool) indicates whether to inject workloadSpread subset
+// 3. suitableSubset(*struct{}) indicates which workloadSpread.Subset does this pod match
+// 4. generatedUID(types.UID) indicates which workloadSpread generate a UID for identifying Pod without a full name.
+func (h *Handler) updateSubsetForPod(ws *appsv1alpha1.WorkloadSpread,
```

Comment:
the second return value (inject) is redundant. There is no case that require injection but without suitable subset.  Outside the function, one can simply check whether the operation is creation and whether subset is not nil